dbms/src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -535,8 +535,7 @@ static JoinPtr tryGetStorageJoin(const ASTTablesInSelectQueryElement & join_elem
 static ExpressionActionsPtr createJoinedBlockActions(const Context & context, const AnalyzedJoin & analyzed_join)
 {
     ASTPtr expression_list = analyzed_join.rightKeysList();
-    auto syntax_result = SyntaxAnalyzer(context).analyze(expression_list,
-                                                         analyzed_join.columnsFromJoinedTable(), analyzed_join.requiredJoinedNames());
```

Comment:
It's interesting that analyzed_join.columnsFromJoinedTable() hasn't been used. SyntaxAnalyzer..removeUnneededColumnsFromSelectClause() is selects-only.